### PR TITLE
add newer go version for environment requirements

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Go uprades to 1.19. The minumum version supported for v6.1.3 and later becomes v1.19.3.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/11512
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
 
 |  编译和构建 TiDB 所需的依赖库   |  版本   |
 |   :---   |   :---   |
-|   Golang  |  1.18.5 及以上版本  |
+|   Golang  |  <ul><li>v6.1.3 之前版本 TiDB：1.18.5 及以上版本</li><li>v6.1.3 及以上版本 TiDB：1.19.3 及以上版本</li></ul> |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:89]:
```suggestion
|   Golang  |  <ul><li>TiDB v6.1.0、v6.1.1 和 v6.1.2：Golang 1.18.5 及以上版本</li><li>TiDB v6.1.3 及以上的 v6.1 版本：Golang 1.19.3 及以上版本</li></ul> |
```

**Review by wuhuizuo — APPROVED** [review]
Please make CI builds green, rest LGTM

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -86,7 +86,7 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
 
 |  编译和构建 TiDB 所需的依赖库   |  版本   |
 |   :---   |   :---   |
-|   Golang  |  1.18.5 及以上版本  |
+|   Golang  |  <ul><li>TiDB v6.1.0、v6.1.1 及 v6.1.2：Golang 1.18 及以上版本</li><li>TiDB v6.1.3：Golang 1.19.3 及以上 版本</li></ul> |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:89]:
```suggestion
|   Golang  |  <ul><li>TiDB v6.1.0、v6.1.1 及 v6.1.2：Golang 1.18 及以上版本</li><li>TiDB v6.1.3：Golang 1.19.3 及以上版本</li></ul> |
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**